profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -14,425 +14,220 @@
-//              verbose will print out each component of the systems power draw
-//              in the sierra architecture. non-verbose will print the node
-//              power.
-// Pre: None
-// Post: Will print an error message if variorum print power fails. No return
-// value.
-std::string variorum_print_power_call() {
-  std::string outputString;
-  json_t* power_obj = json_object();
-  double power_node, power_sock0, power_mem0, power_gpu0;
-  double power_sock1, power_mem1, power_gpu1;
-  int ret;
-  ret = variorum_get_node_power_json(power_obj);
-  if (ret != 0) {
-    return "Print power failed!\n";
-  }
-  // total node measurment
-  power_node = json_real_value(json_object_get(power_obj, "power_node"));
-  const char* hostnameChar =
-      json_string_value(json_object_get(power_obj, "hostname"));
-  std::string hostname(hostnameChar);
-  // print informatin to screen
-  if (verbosePrint) {
-    // socket 1 measurements
-    power_sock0 =
-        json_real_value(json_object_get(power_obj, "power_cpu_socket_0"));
-    power_mem0 =
-        json_real_value(json_object_get(power_obj, "power_mem_socket_0"));
-    power_gpu0 =
-        json_real_value(json_object_get(power_obj, "power_gpu_socket_0"));
-    // socket 2 measurements
-    power_sock1 =
-        json_real_value(json_object_get(power_obj, "power_cpu_socket_1"));
-    power_mem1 =
-        json_real_value(json_object_get(power_obj, "power_mem_socket_1"));
-    power_gpu1 =
-        json_real_value(json_object_get(power_obj, "power_gpu_socket_1"));
-
-    outputString += "HostName " + hostname + "\n";
-    outputString += "Total Node Power: " + std::to_string(power_node);
-    outputString += "\n Socket 1 Power";
-    outputString += "\n CPU Socket 1: " + std::to_string(power_sock0);
-    outputString += "\n Mem Socket 1: " + std::to_string(power_mem0);
-    outputString += "\n GPU Socket 1: " + std::to_string(power_gpu0);
-    outputString += "\n Socket 2 Power";
-    outputString += "\n CPU Socket 2: " + std::to_string(power_sock1);
-    outputString += "\n Mem Socket 2: " + std::to_string(power_mem1);
-    outputString += "\n GPU Socket 2: " + std::to_string(power_gpu1) + "\n";
+  std::ifstream infile(global_filename);
+  if (infile.good()) {
+    infile.close();
+    std::ofstream file(global_filename, std::ios::trunc);
   } else {
-    outputString += hostname + ": " + std::to_string(power_node) + "\n";
+    std::ofstream file(global_filename, std::ios::app);
   }
-
-  return outputString;
 }
 
-// Function: variorum_json_call()
-// Description: function that will call variorum print json and handle the
-// execution errors Pre: None Post: Will print an error message if variorum
-// print json fails. No return value.
-char* variorum_json_call() {
-  int ret;
-  json_t* my_power_obj = NULL;
+void variorum_call() {
+  char* s              = nullptr;
+  json_t* my_power_obj = nullptr;
   my_power_obj         = json_object();
```

Comment:
```suggestion
  json_t* my_power_obj = json_object();
```

Reply:
Fragment from previous version, remove `my_power_obj`. We're using `root` now instead.